### PR TITLE
Implements new notifications in back-office header

### PR DESCRIPTION
**Pull request by maximebiloe**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | With this PR you now have the new notifications system |
| Type? | improvement |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | yes - No more option to enable/disable notifications auto-refresh |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-665 |
| How to test? | Check your notifications on both old and new pages and it should look like screenshot |

capture d ecran 2016-05-05 a 12 28 20

### CODE REVIEW

**Comment by dSkrbic** [comment]
OK for the Q&A, although remains one point: integration of the compagny name in command notifications (when in BtoB).
See comments and screenshots in ticket BOOM-665

File: admin-dev/themes/new-theme/js/header.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 
 export default class Header {
   constructor() {
     $(() => {
       this.initQuickAccess();
       this.initMultiStores();
+      this.initNotificationsToggle();
       this.initSearch();
     });
   }
```
Comment (Shudrum) [inline admin-dev/themes/new-theme/js/header.js]: Maybe we should add some space between methods ?

https://github.com/airbnb/javascript#constructors--tostring

File: admin-dev/themes/new-theme/js/notifications.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      );
+    });
+  } else {
+    eltAppendTo.addClass('empty');
+  }
+}
+
+let setNotificationsNumber = function (id, number) {
+  if (number > 0) {
+    $("#" + id).text(" (" + number + ")");
```
Comment (Shudrum) [inline admin-dev/themes/new-theme/js/notifications.js]: Maybe we should start to use ```$(`#${id}`).text(` (${number}`);```

https://github.com/airbnb/javascript#es6-template-literals

File: admin-dev/themes/new-theme/scss/theme.scss [hunk 3]
```diff
@@ -14,6 +14,7 @@
 @import "components/layout/main_header";
 @import "components/layout/nav_bar";
 @import "components/category_tree";
+@import "components/_notifications";
```
Comment (Shudrum) [inline admin-dev/themes/new-theme/scss/theme.scss]: @import "components/notifications";

File: admin-dev/themes/new-theme/template/components/layout/notifications_center.tpl [hunk 4]
```diff
@@ -1,12 +1,60 @@
...
+    <i class="material-icons">notifications_none</i>
     {* TODO: REFACTORING THIS IDENTIFIER *}
     <span id="orders_notif_value" class="count">0</span>
   </div>
   <div class="dropdown-menu dropdown-menu-right">
     <div class="notifications">
-      {l s='No new orders have been placed on your shop.'}
+      <ul class="nav nav-tabs" role="tablist">
+        <li class="nav-item">
+          <a class="nav-link active" data-toggle="tab" data-type="order" href="#orders-notifications" role="tab" id="orders-tab">{l s='Orders[1][/1]' tags=['<span id="_nb_new_orders_">']}</a>
```
Comment (Shudrum) [inline admin-dev/themes/new-theme/template/components/layout/notifications_center.tpl]: 120 chars max please

Comment (Shudrum) [inline admin-dev/themes/new-theme/template/components/layout/notifications_center.tpl]: And try to follow the html attributes order.

File: admin-dev/themes/new-theme/template/components/layout/notifications_center.tpl [hunk 5]
```diff
@@ -1,12 +1,60 @@
 <div class="notification-center dropdown">
-  <div class="notification dropdown-toggle" data-toggle="dropdown">
-    <i class="material-icons">notifications</i>
+  <div class="notification dropdown-toggle">
+    <i class="material-icons">notifications_none</i>
     {* TODO: REFACTORING THIS IDENTIFIER *}
```
Comment (Shudrum) [inline admin-dev/themes/new-theme/template/components/layout/notifications_center.tpl]: Maybe we can remove this todo ?
_(And its mine ^^)_

File: classes/controller/AdminController.php [hunk 6]
```diff
@@ -2722,7 +2748,11 @@ public function init()
             'current' => self::$currentIndex,
             'token' => $this->token,
             'host_mode' => defined('_PS_HOST_MODE_') ? 1 : 0,
-            'stock_management' => (int)Configuration::get('PS_STOCK_MANAGEMENT')
+            'stock_management' => (int)Configuration::get('PS_STOCK_MANAGEMENT'),
+            'abandoned_cart_url' => $this->context->link->getAdminLink('AdminCarts').'&action=filterOnlyAbandonedCarts',
+            'no_order_tip' => $this->getNotificationTip('order'),
+            'no_customer_tip' => $this->getNotificationTip('customer'),
+            'no_customer_message_tip' => $this->getNotificationTip('customer_message')
```
Comment (Shudrum) [inline classes/controller/AdminController.php]: Missed the last coma ^^

**Comment by maximebiloe** [comment]
@Shudrum thank you for review
I made the changes, now waiting for tests!

**Pull request merged**